File: openbis_api/source/java/ch/ethz/sis/openbis/generic/asapi/v3/dto/rights/get/GetRightsOperation.java
/*
 * Copyright 2019 ETH Zuerich, SIS
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *      http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.
 */

package ch.ethz.sis.openbis.generic.asapi.v3.dto.rights.get;

import java.util.List;

import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.get.GetObjectsOperation;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.common.id.IObjectId;
import ch.ethz.sis.openbis.generic.asapi.v3.dto.rights.fetchoptions.RightsFetchOptions;
import ch.systemsx.cisd.base.annotation.JsonObject;

/**
 * @author Franz-Josef Elmer
 */
@JsonObject("as.dto.rights.get.GetRightsOperation")
public class GetRightsOperation extends GetObjectsOperation<IObjectId, RightsFetchOptions>
{
    private static final long serialVersionUID = 1L;

    @SuppressWarnings("unused")
    private GetRightsOperation()
    {
    }

    public GetRightsOperation(List<? extends IObjectId> ids, RightsFetchOptions fetchOptions)
    {
        super(ids, fetchOptions);
    }
}

File: openbis/source/java/ch/systemsx/cisd/openbis/public/resources/api/v3/as/dto/rights/get/GetRightsOperation.js
define([ "stjs", "as/dto/common/get/GetObjectsOperation" ], function(stjs, GetObjectsOperation) {
	var GetRightsOperation = function(objectIds, fetchOptions) {
		GetObjectsOperation.call(this, objectIds, fetchOptions);
	};
	stjs.extend(GetRightsOperation, GetObjectsOperation, [ GetObjectsOperation ], function(constructor, prototype) {
		prototype['@type'] = 'as.dto.rights.get.GetRightsOperation';
		prototype.getMessage = function() {
			return "GetRightsOperation";
		};
	}, {});
	return GetRightsOperation;
})
